pkg/cmd/roachtest/tests/failover.go

```diff
@@ -70,6 +77,74 @@ func registerFailover(r registry.Registry) {
 	}
 }
 
+// 5 nodes fully connected. Break the connection between a pair of nodes 4 and 5
+// while running a workload against nodes 1 through 3. Before each disconnect,
+// move all the leases to nodes 4 and 5 in a different pattern.
+func runDisconnect(ctx context.Context, t test.Test, c cluster.Cluster) {
```

Comment:
Let's try to be more specific about the scenario we're testing here, to have tigher controls on the failure mode and required solution (which also avoids unintended flake). We specifically want to test a partial partition between a user range leaseholder and the liveness range leaseholder -- no more, no less. I suggest we place ranges as follows:

- System ranges: n1,n2,n3
- Liveness range: n1,n2,n4 with lease on n4
- User ranges: n2,n3,n5 with leases on n5
- SQL gateways: n1,n2,n3

This is almost what we have here already, but we should use zone configurations to ensure we e.g. don't end up with the meta range on n4 and that starts messing with things (we'll unfortunately also want to move ranges manually to avoid having to wait for the replicate queue). This also ensures that `WaitFor3XReplication` actually waits for all upreplication, since it won't wait for e.g. 4x or 5x replication which could cause this to flake.